.circleci/real_config.yml

```diff
@@ -1432,12 +1437,21 @@ commands:
             - run: make -C agent check
 
   make-package:
+    parameters:
+      dryrun:
+        type: boolean
+        default: false
     steps:
       - attach_workspace:
           at: .
       - run:
           no_output_timeout: 30m
-          command: make package
+          command: |
```

Comment:
https://circleci.com/docs/configuration-reference/#the-when-step
https://support.circleci.com/hc/en-us/articles/360043638052-Conditional-steps-in-jobs-and-conditional-workflows
https://hpe-aiatscale.atlassian.net/wiki/spaces/~71202061ef532da5624595b2ac8d30dd426710/pages/1737261083/Circle+CI+Docs+Reference+Sheet

Reply:
I'm not sure I follow. `when` clauses allow you to run jobs conditionally, but I have two separate branches I have to take, and it looks like jobs only support a single `when` clause. (I could maybe use `when` and `unless` together, but I've never been a fan of `unless` ever since my Perl days; always having to invert the conditional can be tricky to reason about.)

Reply:
two mistakes here
* when can be used at the workflow level and the step level, not the job level
* you can nest conditions
https://support.circleci.com/hc/en-us/articles/360043638052-Conditional-steps-in-jobs-and-conditional-workflows

Reply:
Okay, I see now. I think keeping the conditional inside the step itself makes it easier to parse what this is doing. I'm also not keen on using `unless` as an `else` clause, because it feels like it's encoding the decision logic in a more implicit way, instead of an explicit shell if/else block. Plus, it's easier to move or delete one of these, and have the meaning of the job change, instead of having a single step that would probably raise a syntax error if it were bumped. What would we gain from this change?